### DISCUSSION TITLE
Sulci mesh

### DISCUSSION THREAD

**Discussion by njanjs** [brainvisa/brainvisa.github.io#144] [Q&A]
Hello! I have generated cervical sulci graphs using BrainVisa/Morphologist pipeline. How could I export sulci meshes from these graphs as I would like to further use sulci segmentation in 3D Slicer?

Thanks!

**Comment by denisri** [comment]
Hi,
First, I want to stress out that meshes in sulcal graphs in Morphologist are generated primarily for visualization purposes. They are smoothed and decimated, thus are only an approximation of the sulci segmentation. If you intent to perform any processing on sulci, you'd better use the skeletons, which are defined as labeled voxels in a 3D image.
Now if you want to display the meshes in 3D slicer, they are in a GIFTI file located inside the `.data` directory nearby the graph `.arg` file. It is generally named `aims_Tmtktri.gii`. All vertices' meshes are concatenated in this file, using the vertex index as timestep in the GIFTI file. There are probably tools to separate them in multiple files, or you can use the following command in brainvisa:
```
AimsGraphConvert -l <input_graph.arg> <new_graph.arg>
```
(don't use the same name as `input_graph` and `output_graph`)
This will convert the graph structure and storage so that meshes and voxels are stored in separate files in the `.data` directory of the new graph. The drawback is that it produces many files there (several thousands), but every `.gii` file is a mesh for a vertex of the sulcal graph.
In 3D slicer I guess you would also have to deal with coordinates systems issues (meshes will likely appear inverted, seen from the inside or black, and at the wrong place). You can use the command `AimsMeshTransform` to apply a transformation to MNI space, for instance, using the transformation files in Morphologist, found in `<center>/<subject>/t1mri/default_acquisition/registration/RawT1-<subject>_default_acquisition_TO_Talairach-MNI.trm`.
Denis

**Reply by njanjs** [reply]
Thanks a lot for your reply, Denis!

I am working on optimisation of electrode trajectories in minimally invasive neurosurgical procedures and in order to take into account a constraint of maintaining the distance between electrodes and sulci, I wanted to work with a sulci mask. However I only managed to generate graph files in brainvisa and I am a bit stuck making use of them.

Could you advise me of what would be a data format suitable for my problem? I did not understand  how to find the skeletons (this seems like a useful format for what I need).

Also, "aims_Tmtktri.gii" file is not loadable in 3D Slicer:
"Error: Loading /Users/anja/Desktop/DataVM/subjects/SCH-IRM/t1mri/default_acquisition/default_analysis/folds/3.1/LSCH-IRM.data/aims_Tmtktri.gii -  load failed."

Thanks a lot for your time and help!

Anja

**Comment by denisri** [comment]
The "skeleton" is an image of the voxels making the sulci shapes. They are named "skeleton" because they are built using a skeletonization algorithm which make the sulci surface 1 voxel thick. It is a 3D image with the same size and resolution as the original T1 MRI, one of the Morphologist outpouts (there is one for the left hemisphere, and one for the right hemisphere, actually). If I understand your problem it would make a possibly good input for you.
"load failed" is not really informative... I am not used to 3D slicer but I'd think it could read Gifti format meshes. But perhaps the problem for it is because the Gifti file you used contains several meshes, one per sulci graph vertex (as I explained above). Converting as I explained might help somewhat, but I'm still not sure meshes are the best objects for you...
Denis

**Reply by njanjs** [reply]
Hi Denis,

I apologise for a late reply. Thanks a lot for your help.
How could I align the skeleton with the corresponding t1mri? 
I am having trouble registering segmentation results from brainvisa with the origin t1mri.

Thanks a lot for your time
Anja

**Comment by denisri** [comment]
The skeleton image is in the same space as the input T1 MRI, it should have the same size (dimensions, voxel sizes) as the T1 and should be directly superimposable.
It is possible that the skeleton image does not always carry the transformation info (I'm not totally sure of if). If so it is easy to fix.
